databricks_cli/pipelines/api.py

```diff
@@ -61,6 +60,15 @@ def get(self, pipeline_id, headers=None):
     def reset(self, pipeline_id, headers=None):
         self.client.reset(pipeline_id, headers)
 
+    def _upload_libraries_and_update_spec(self, spec):
+        spec = copy.deepcopy(spec)
```

Comment:
Why'd you add this thing?

Reply:
The deepcopy makes sure that `_upload_libraries_and_update_spec` is a pure function. It's not really necessary here but it make sure that the spec passed through the arguments is not modified.